bcipy/display/components/layout.py

```diff
@@ -34,17 +35,85 @@ def vertical(cls):
         return [Alignment.CENTERED, Alignment.TOP, Alignment.BOTTOM]
 
 
+# Positioning functions
+def above(y_coordinate: float, amount: float) -> float:
+    """Returns a new y_coordinate value that is above the provided value
+    by the given amount."""
+    assert amount >= 0, 'Amount must be positive'
+    return y_coordinate + amount
+
+
+def below(y_coordinate: float, amount: float) -> float:
+    """Returns a new y_coordinate value that is below the provided value
+    by the given amount."""
+    assert amount >= 0, 'Amount must be positive'
+    return y_coordinate - amount
+
+
+def right_of(x_coordinate: float, amount: float) -> float:
+    """Returns a new x_coordinate value that is to the right of the
+    provided value by the given amount."""
+    assert amount >= 0, 'Amount must be positive'
+    return x_coordinate + amount
+
+
+def left_of(x_coordinate: float, amount: float) -> float:
+    """Returns a new x_coordinate value that is to the left of the
+    provided value by the given amount."""
+    assert amount >= 0, 'Amount must be positive'
+    return x_coordinate - amount
+
+
+def envelope(pos: Tuple[float, float],
+             size: Tuple[float, float]) -> List[Tuple[float, float]]:
+    """Compute the vertices for the envelope of a shape centered at pos with
+    the given size."""
+    width, height = size
+    half_w = width / 2
+    half_h = height / 2
+    return [(left_of(pos[0], half_w), above(pos[1], half_h)),
+            (right_of(pos[0], half_w), above(pos[1], half_h)),
+            (right_of(pos[0], half_w), below(pos[1], half_h)),
+            (left_of(pos[0], half_w), below(pos[1], half_h))]
+
+
+def scaled_size(height: float,
+                window_size: Tuple[float, float],
+                units: str = 'norm') -> Tuple[float, float]:
+    """Scales the provided height value to reflect the aspect ratio of a
+    visual.Window. Used for creating squared stimulus. Returns (w,h) tuple"""
+    if units == 'height':
```

Comment:
We should standardize to one unit type and assert that in our display... I use height in the inquiry preview, but it might be worth a refactor.

Reply:
In practice I have been using norms for everything. When I finish working on the ticket for matrix row heights it should be easy to refactor Inquiry Preview to use that function to compute positions in 'norm' units.